src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -421,8 +421,9 @@ void SoundRecorder::setChannelCount(unsigned int channelCount)
         {
             m_impl->channelMap = {SoundChannel::Mono};
         }
-        else if (channelCount == 2)
+        else
         {
+            assert(channelCount == 2);
```

Comment:
I don't particularly like this pattern, as it move the branch condition into the assert and mixes in the body two different concerns: 1) Error checking for the input 2) actual body for the given condition

Reply:
It's the same principle as above -- that condition is guaranteed to be true due to previous control flow. The assert makes that obvious to the reader.